Taskfile.yml

```diff
@@ -69,12 +69,19 @@ tasks:
       - workspace/diff-engine-wasm/engine/target/**/*
       - workspaces/*/build/**/*
 
+  workspaces:test:
+    desc: Test Yarn workspaces
+    cmds:
+      - yarn wsrun --stages --report --fast-exit --exclude-missing ws:test
+
   workspaces:build:ci:
     desc: CI workflow
     cmds:
       - task: workspaces:setup
       - task: workspaces:clean
       - task: workspaces:build
+      - cargo build
```

Comment:
@JaapRood seems like this is required for the cli-shared tests. I'm not clear on where we're supposed to be running cargo build for what. Can you clarify? Anyway, I think we can get your diff-engine reorg merged and then apply most of the changes from here on top

Reply:
Not really sure, especially as `workspaces:build` should include the building of any native binaries 🤔. `cli-shared` does test the native <> Node.js interface, so there having to be a build seems right. But why the separate commands is necessary needs further investigation. 

As you said, probably better based on top of the re-organisation, as there's some chance this has solved itself already.